pkg/eventemitter/eventtypes.go

```diff
@@ -0,0 +1,25 @@
+/*
+Copyright 2023 The KEDA Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eventemitter
+
+const (
+	// ScaledObjectReadyType is for event when a new ScaledObject is ready
+	ScaledObjectReadyType = "keda.scaledobject.ready.v1"
+
+	// ScaledObjectCheckFailedType is for event when ScaledObject validation check fails
+	ScaledObjectCheckFailedType = "keda.scaledobject.checkfailed.v1"
```

Comment:
Can you elaborate on what these 2 are? They are not in https://github.com/kedacore/keda/issues/3522 so I'm guessing they are in addition to these?

Reply:
These two are all related to manage scaledobject resources. I think "ready" is more accurate than keda.scaledobject.created.v1 because it will be emitted after ready checking. And the "checkfailed" is about creating scaledobject failed scenario. These two are the easy events that can trigger so I added these two. What do you think? @tomkerkhove

Reply:
Frankly, I think they are too technical and created/failed make more sense if they are covering the same scenario IMO.

The goal is to have them as events that anyone can extend and can be processed outside of Kubernetes context (which they might not even know that it's there) so I believe this can be confusing.

What do you think @kedacore/keda-maintainers

Reply:
I don't have an strong opinion, but maybe we can have all of them. As user, I want to know that the resource has been created but also if the resource is ready or not. About if the check has passed or not, I don't know if it's necessary or not but what does it happen if the creation fails? Do we send any other?

Reply:
I plan to send other failed events. Maybe those failed events about scaledobject can combine into one "keda.scaledobject.failed.v1"event other than "keda.scaledobject.checkfailed.v1"?

Reply:
If we have both, then I honestly don't really know what the value is of both? Every failed scaled object will generate two events for same approach?

Reply:
Do you mean every failed scaled object will generate "created/ready" and "failed" events? Or two "failed" events will be generated?

Reply:
I think having 1 failed is enough. If it failed, it can be because of multiple reasons (validation today, other scenarios in future). Having a created is a must, having a ready is a fine addition to me but not a must.

Reply:
I agree. 1 failed is enough. Do you have a good place to trigger this failed event? @JorTurFer

Reply:
Maybe it the reconciliation loop? I haven't checked it yet and I'm just suggesting it from memory, but the reconciliation loop should have a place where we return an error in case of errors (but maybe I'm wrong, today has been a long day)